docs/Planner.md

```diff
@@ -0,0 +1,108 @@
+(left TFs in each graph) without further surrounding information, e.g. the table function (TF) `nested_fields(q2)` on the
+querying side versus `nested_fields(qa)` on the querying side, it is inherently impossible to reason if these two table
+functions match without knowing if `q2` is the same as `qa` or not. We can only know that once we reach a higher level
+during matching. In the example, it is correct to conclude `q2` is equivalent to `qa` once we match the select expressions
+containing `q1` respectively `qa`. Therefore, for the left of the two table functions in the respective graphs the table
+functions do match in that case. However, that would not be the case for the right one on the query graph side (green) side
+compared with the left one on the index scan graph side (red). We again wouldn’t know (given that they let’s say both do
+a nested_fields() operation) that `nested_fields(q4)` is not equivalent to `nested_fields(qa)` when we attempt to match
+the table functions as we don’t understand yet if there is a correspondence between `q4` and `qa` (which there is not).
+As a consequence of this observation we need to incorporate the assumptions we have about the equivalence of correlation
+sources into our partial match structure and see if they hold up later. `nested_fields(q2)` can be matched to
+`nested_fields(qa)` under the assumption that `q2` is equivalent to `qa`. Later it can be proven that `q2` is in fact
+equivalent to `qa` and the assumption holds. The opposite can be done for our example that eventually would not produce
+a match. `nested_fields(q4)` can be matched to `nested_fields(qa)` under the assumption that `q4` is equivalent to `qa`.
+Later on we will realize and disprove that but when we only try to match bottom up that decision will have to be deferred.
+In the remainder we will use the phrase: ... partial match under mapping `q4 → qa`...
+On the way up from the leaves to the root of the respective query DAGs we can match more and more of the graphs if
+that’s at all possible.
+For a non-leaf expression we will only be able to match it to the corresponding expression on the index scan graph side
+if all children are partially matched under compatible correlation mappings. If there is no such mapping where a common
+partial match may exist or the current expression at hand is structurally not equivalent (or more general subsuming),
+the current expression cannot be matched and matching for this particular index scan graph is impossible for parents
+of this expression. If, on the other hand, we reach the top-most expression on the index scan graph, we matched the index
+scan graph entirely, and can replace the index scan graph with an index scan operator.
+While we match query graph to index scan graph we may encounter a predicate (or a some other sort of scalar expression)
+that is specific on the query side, i.e. `x > 5` in the example and it needs to match a comparison range on the index
```

Comment:
```suggestion
that is specific on the query side, e.g. `x > 5` in the example, that needs to match a comparison range on the index
```

I suppose there's an argument for "i.e." in this case, though I think it's more of an "e.g.".